status_test.go

```diff
@@ -112,3 +119,61 @@ func TestStatusHandlerReloading(t *testing.T) {
 		t.Error("status should return 200 during reload")
 	}
 }
+
+func TestStatusTarget2XX(t *testing.T) {
+	// Create a stub status target that returns a 200 status code.
+	statusTarget := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		w.WriteHeader(200)
+	}))
+	defer statusTarget.Close()
+
+	response := httptest.NewRecorder()
+	handler := newStatusHandler(func() (net.Conn, error) {
+		u, _ := url.Parse(statusTarget.URL) // NOTE: I tried using statusTarget.Config.Addr instead, but it wasn't set.
```

Comment:
Just wanted to call this out because I really don't like this level of complexity in the test.